pkg/extender/extender.go

```diff
@@ -1043,38 +1032,41 @@ func (e *Extender) updateVirtLauncherPodPrioritizeScores(
 		}
 	} else {
 		driverNodes = volume.RemoveDuplicateOfflineNodes(driverNodes)
+		isExistingPodUsingLocallyAttachedVolume := e.isPodUsingLocallyAttachedVolume(podBeingLiveMigrated, volInfo, driverNodes)
+		if isExistingPodUsingLocallyAttachedVolume {
+			storklog.PodLog(pod).Infof("Existing pod is using locally attached volume")
+		}
+
 		for _, dnode := range driverNodes {
 			for _, knode := range args.Nodes.Items {
-				var score int64
-				isUpdated := false
+				// Initialize with score to with score for remote node
+				// It would be updated if the node being score is a replica node
+				score := remoteNodeScore
 				if volume.IsNodeMatch(&knode, dnode) {
-					isUpdated = false
 					// Score replica nodes
-					for _, dataNodes := range volInfo.DataNodes {
-						if dataNodes == dnode.StorageID {
+					for _, dataNode := range volInfo.DataNodes {
+						if dataNode == dnode.StorageID {
+							// Current node has the volume replica
 							score = replicaNodeScore
-							isUpdated = true
-							break
-						}
-					}
-
-					if preferLocalAttachment && volInfo.AttachedOn != "" {
-						// Update score of node for volume attachment
-						for _, nodeIP := range dnode.IPs {
-							if nodeIP == volInfo.AttachedOn {
-								// Local attachment
-								storklog.PodLog(pod).Debugf("Updating score of node: %v for local attachment", dnode.StorageID)
-								score = localNodeScore
-								isUpdated = true
+							nodeHasAttachedVolume := e.nodeHasAttachedVolume(dnode, volInfo)
+
+							if podBeingLiveMigrated != nil {
+								// LiveMigration in progress
+								// If existing pod is not using attached volume and current node has attached volume, assign a higher score to it
+								if !isExistingPodUsingLocallyAttachedVolume && nodeHasAttachedVolume {
```

Comment:
wondering if we can always give higher score to the node with attached volume. If pod is running on that node, it will not be chosen anyways by KubeVirt since the pod is being migrated out of that node.

Reply:
We could do that based on our observations that the node on which current pod is running is never going to be in the input list of nodes.

But let's say Kubevirt didn't cordon the node(or there is some other thing that cause the current node to be in the input set of nodes to be filtered), then as we know that a Pod is already running on the node with same VMI reference we shouldn't be choosing the node it even if we have a local volume on it. All we are saying here is if we are in a migration scenario and the original Pod is still running then do not give any extra preference based on local attachment which is logically correct irrespective of the input set of nodes give to the filter/prioritize function.
Let me know if you disagree here.

Reply:
Yes, the code looks correct as it is. I was just wondering if we could simplify it by taking into account how KubeVirt works. But I am fine with leaving it as it is.

Reply:
I will merge this PR as is to park the current implementation. Will send another PR for simplifying this.